src/inmanta/agent/executor.py

```diff
@@ -116,7 +115,7 @@ def blueprint_hash(self) -> str:
     def __eq__(self, other: object) -> bool:
         if not isinstance(other, EnvBlueprint):
             return False
-        return self.blueprint_hash() == other.blueprint_hash()
+        return (self.pip_config, set(self.requirements)) == (other.pip_config, set(other.requirements))
```

Comment:
Technically, requirement order may affect install behavior. But to be honest I think we may already mess with the order in other places, not sure.

Reply:
> we don't take it into account anywhere.

Do you mean we don't make an effort to preserve order anywhere?

I'll elaborate a bit on what I meant. Perhaps you already understood exactly this, but I'd rather be sure we're on the same page.

I meant that when we delegate to pip, it might give different install behavior if the order is different. Yet we consider the blueprints equal if they have the same dependencies, regardless of order. So suppose a user would change the dependency order in an attempt to address installation issues, it would not be picked up by the env manager here.

Reply:
I don't think the code for merging requirements is intended tot preserve order (it might)

but still, I can make it take order into account.